ctest (58+60,/67 cmake_import_configure) fails when build with -D JSON_Install:BOOL=OFF because of missing nlohmann_jsonTargets.cmake

#### What is the issue you have?

```
Start 58: cmake_import_configure
58/67 Test #58: cmake_import_configure .......................***Failed    2.07 sec
...
-- Found nlohmann_json: build/json/nlohmann_jsonConfig.cmake (found version "3.10.0")
CMake Error at build/json/nlohmann_jsonConfig.cmake:6 (include):
  include could not find requested file:

    build/json/nlohmann_jsonTargets.cmake
Call Stack (most recent call first):
  CMakeLists.txt:5 (find_package)
```
This is because in CMakeLists.txt export ... NLOHMANN_JSON_CMAKE_PROJECT_TARGETS_FILE is called only in case of JSON_Install is true.

#### Please describe the steps to reproduce the issue.

1. cmake .. -D JSON_Install:BOOL=OFF -D JSON_ImplicitConversions:BOOL=OFF -D JSON_BuildTests:BOOL=ON
2. cmake --build . --config Release
3. ctest -C Release --output-on-failure

#### What is the expected behavior?

Tests should not fail when using flags -D JSON_Install:BOOL=OFF -D JSON_ImplicitConversions:BOOL=OFF -D JSON_BuildTests:BOOL=ON

#### And what is the actual behavior instead?

Some tests fail.

#### Which compiler and operating system are you using?

- Compiler:  Visual Studio 2019 16.11.1 Professional
- Operating system:  Windows 10 21H1

#### Which version of the library did you use?

- [x] latest release version 3.10.0 (zip file)
- [ ] other release - please state the version: ___
- [ ] the `develop` branch

#### If you experience a compilation error: can you [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests)?

- [ ] yes
- [x] no - please copy/paste the error message below

I can confirm this.

Full output:

```
58/67 Test #58: cmake_import_configure .......................***Failed    0.41 sec
-- The CXX compiler identification is AppleClang 12.0.5.12050022
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found nlohmann_json: /Users/niels/Documents/repositories/json/build/nlohmann_jsonConfig.cmake (found version "3.10.0") 
CMake Error at /Users/niels/Documents/repositories/json/build/nlohmann_jsonConfig.cmake:6 (include):
  include could not find requested file:

    /Users/niels/Documents/repositories/json/build/nlohmann_jsonTargets.cmake
Call Stack (most recent call first):
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!
See also "/Users/niels/Documents/repositories/json/build/test/cmake_import/CMakeFiles/CMakeOutput.log".
See also "/Users/niels/Documents/repositories/json/build/test/cmake_import/CMakeFiles/CMakeError.log".

      Start 59: cmake_import_build
Failed test dependencies: cmake_import_configure
59/67 Test #59: cmake_import_build ...........................***Not Run   0.00 sec
      Start 60: cmake_import_minver_configure
60/67 Test #60: cmake_import_minver_configure ................***Failed    0.46 sec
-- The CXX compiler identification is AppleClang 12.0.5.12050022
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found nlohmann_json: /Users/niels/Documents/repositories/json/build/nlohmann_jsonConfig.cmake (found suitable version "3.10.0", minimum required is "3.2.0") 
CMake Error at /Users/niels/Documents/repositories/json/build/nlohmann_jsonConfig.cmake:6 (include):
  include could not find requested file:

    /Users/niels/Documents/repositories/json/build/nlohmann_jsonTargets.cmake
Call Stack (most recent call first):
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!
See also "/Users/niels/Documents/repositories/json/build/test/cmake_import_minver/CMakeFiles/CMakeOutput.log".
See also "/Users/niels/Documents/repositories/json/build/test/cmake_import_minver/CMakeFiles/CMakeError.log".

      Start 61: cmake_import_minver_build
Failed test dependencies: cmake_import_minver_configure
61/67 Test #61: cmake_import_minver_build ....................***Not Run   0.00 sec
...
94% tests passed, 4 tests failed out of 67

Label Time Summary:
all                 = 639.60 sec*proc (56 tests)
git_required        =   6.52 sec*proc (2 tests)
not_reproducible    =  22.95 sec*proc (10 tests)

Total Test time (real) = 677.98 sec

The following tests FAILED:
	 58 - cmake_import_configure (Failed)
	 59 - cmake_import_build (Not Run)
	 60 - cmake_import_minver_configure (Failed)
	 61 - cmake_import_minver_build (Not Run)
Errors while running CTest
```

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -115,8 +115,12 @@ add_dependencies(json_unit download_test_data)
 # Test the generated build configs
 #############################################################################
 
-add_subdirectory(cmake_import)
-add_subdirectory(cmake_import_minver)
+# these tests depend on the generated file nlohmann_jsonConfig.cmake
+if (JSON_Install)
+    add_subdirectory(cmake_import)
+    add_subdirectory(cmake_import_minver)
+endif()
+
 add_subdirectory(cmake_add_subdirectory)
 add_subdirectory(cmake_fetch_content)
 add_subdirectory(cmake_target_include_directories)

